### PR TITLE
Feature/holo 751 query nft by ipfs cid

### PR DESCRIPTION
**Pull request by alexanderattar**
## Describe Changes

- Adds new query to get nft by ipfs CID
- Adds validation function for CID
- Update GQL calls to API to get NFTs by ipfs CID instead of tx
- Sets tx in the query to undefined so it only finds the new NFTs that haven't been minted yet
- Ran prettier

```
[2023-01-18T20:59:27.631Z] [Indexer] [Ethereum Goerli] A new block has been mined. New block height is [8334634] ⛏
[2023-01-18T20:59:28.147Z] [Indexer] [Ethereum Goerli] [8334634] Getting block 🔍
[2023-01-18T20:59:28.564Z] [Indexer] [Ethereum Goerli] [8334634] Block retrieved 📥
[2023-01-18T20:59:28.565Z] [Indexer] [Ethereum Goerli] [8334634] Found 1 interesting transactions
[2023-01-18T20:59:28.565Z] [Indexer] [Ethereum Goerli] [8334634] [6f48a570] Processing transaction 0x27fdebecc7508b1b77356516b680f1d0bc2be7e23e92618c4cd8ed5374957d23 at block 8334634
[2023-01-18T20:59:28.566Z] [Indexer] [Ethereum Goerli] [8334634] [6f48a570] handleMintEvent https://goerli.etherscan.io/tx/0x27fdebecc7508b1b77356516b680f1d0bc2be7e23e92618c4cd8ed5374957d23
[2023-01-18T20:59:28.763Z] [Indexer] [Ethereum Goerli] [8334634] [6f48a570] Checking for mint details
[2023-01-18T20:59:28.765Z] [Indexer] [Ethereum Goerli] [8334634] [6f48a570] updateMintedERC721
[2023-01-18T20:59:28.765Z] [Indexer] [Ethereum Goerli] [8334634] [6f48a570] Indexer identified a minted an ERC721 NFT. Holographer minted a new NFT on EthereumTestnetGoerli at address 0xDC39EBF3746F1E0fb7EF2bdf47C44360579d0A29. The ID of the NFT is 0x0000000000000000000000000000000000000000000000000000000000000008. Account that minted the nft is 0xa069D65dFeA771Ff6594194F309A7697c8946B24
[2023-01-18T20:59:28.765Z] [Indexer] [Ethereum Goerli] [8334634] [6f48a570] Checking if contract 0xDC39EBF3746F1E0fb7EF2bdf47C44360579d0A29 is on registry ...
[2023-01-18T20:59:28.765Z] [Indexer] [Ethereum Goerli] [8334634] [6f48a570] registry Contract address = 0xae27815bcf7cca7191cb55a6b86576aedc462bbb
[2023-01-18T20:59:29.009Z] [Indexer] [Ethereum Goerli] [8334634] [6f48a570] isHolographable = true with type = boolean
[2023-01-18T20:59:29.009Z] [Indexer] [Ethereum Goerli] [8334634] [6f48a570] Contract 0xDC39EBF3746F1E0fb7EF2bdf47C44360579d0A29 is in registry at develop
[2023-01-18T20:59:29.009Z] [Indexer] [Ethereum Goerli] [8334634] [6f48a570] Attaching CXIP ERC721 Contract to the contract address 0xDC39EBF3746F1E0fb7EF2bdf47C44360579d0A29
[2023-01-18T20:59:29.010Z] [Indexer] [Ethereum Goerli] [8334634] [6f48a570] Calling the tokenURI function for tokenId 0x0000000000000000000000000000000000000000000000000000000000000008
[2023-01-18T20:59:29.289Z] [Indexer] [Ethereum Goerli] [8334634] [6f48a570] Token URI is ipfs://QmVwgMZuweHvCdphpesioubbvFLN86DA47rsf97zhEfDjg/metadata.json
[2023-01-18T20:59:29.290Z] [Indexer] [Ethereum Goerli] [8334634] [6f48a570] IPFS CID is QmVwgMZuweHvCdphpesioubbvFLN86DA47rsf97zhEfDjg
[2023-01-18T20:59:29.290Z] [Indexer] [Ethereum Goerli] [8334634] [6f48a570] Validating IPFS CID QmVwgMZuweHvCdphpesioubbvFLN86DA47rsf97zhEfDjg
[2023-01-18T20:59:29.290Z] [Indexer] [Ethereum Goerli] [8334634] [6f48a570] Sending minted nft with IPFS CID QmVwgMZuweHvCdphpesioubbvFLN86DA47rsf97zhEfDjg and tx 0x27fdebecc7508b1b77356516b680f1d0bc2be7e23e92618c4cd8ed5374957d23 job to DBJobManager
[2023-01-18T20:59:29.574Z] [Indexer] [Ethereum Goerli] [8334634] Block procesing complete ✅
Checking if jobs exist for timestamp 1674075564...
[2023-01-18T20:59:30.570Z] [Indexer] [Ethereum Goerli] [8334634] [6f48a570] Processing job...
[2023-01-18T20:59:30.570Z] [Indexer] [Ethereum Goerli] [8334634] [6f48a570] API: Requesting to update NFT with IPFS CID QmVwgMZuweHvCdphpesioubbvFLN86DA47rsf97zhEfDjg and transaction hash 0x27fdebecc7508b1b77356516b680f1d0bc2be7e23e92618c4cd8ed5374957d23
[2023-01-18T20:59:30.571Z] [Indexer] [Ethereum Goerli] [8334634] [6f48a570] Sending query request "\n query($nftByIpfsCid: GetNftByIpfsCidInput!) {\n nftByIpfsCid(nftInput: $nftByIpfsCid) {\n id\n tx\n status\n chainId\n }\n }\n " with props {"nftByIpfsCid":{"cid":"QmVwgMZuweHvCdphpesioubbvFLN86DA47rsf97zhEfDjg","tx":null}}
[2023-01-18T20:59:30.601Z] [Indexer] [Ethereum Goerli] [8334634] [6f48a570] [I38AZvcuRt+kzDvG3Qb57Q/0] Query response {"nftByIpfsCid":{"id":"68818026-d687-4efc-a98b-303af598d654","tx":null,"status":"MINTING","chainId":5}}
[2023-01-18T20:59:30.602Z] [Indexer] [Ethereum Goerli] [8334634] [6f48a570] Successfully found NFT with tx 0x27fdebecc7508b1b77356516b680f1d0bc2be7e23e92618c4cd8ed5374957d23
[2023-01-18T20:59:30.602Z] [Indexer] [Ethereum Goerli] [8334634] [6f48a570] API: Requesting to update NFT with null and id 68818026-d687-4efc-a98b-303af598d654
[2023-01-18T20:59:30.602Z] [Indexer] [Ethereum Goerli] [8334634] [6f48a570] Sending mutation request "\n mutation($updateNftInput: UpdateNftInput!) {\n updateNft(updateNftInput: $updateNftInput) {\n id\n tx\n status\n chainId\n }\n }\n " with props {"updateNftInput":{"id":"68818026-d687-4efc-a98b-303af598d654","tx":"0x27fdebecc7508b1b77356516b680f1d0bc2be7e23e92618c4cd8ed5374957d23","status":"MINTED","chainId":5}}
[2023-01-18T20:59:30.619Z] [Indexer] [Ethereum Goerli] [8334634] [6f48a570] [1RZH5/bCRHO5tl8C/J510A/0] Successfully updated NFT with transaction hash 0x27fdebecc7508b1b77356516b680f1d0bc2be7e23e92618c4cd8ed5374957d23
```

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [x] Code styles have been enforced
- [x] I have checked eslint

### CODE REVIEW

File: .github/workflows/cicd_develop_staging.yaml [hunk 1]
```diff
@@ -36,7 +36,7 @@ env:
   #
   #######################################
   STG_HOLOGRAPH_INDEXER_HELM_CHART_VERSION: 0.0.69
-  INDEXER_RELEASE_NAME: indexer-dev  # format -> [release_name]-indexer-[env]
+  INDEXER_RELEASE_NAME: indexer-dev # format -> [release_name]-indexer-[env]
```
Comment (sogoiii) [inline .github/workflows/cicd_develop_staging.yaml:39]: @alexanderattar why are these files modified? I believe whatever is in develop should be the correct version.

File: src/commands/indexer/index.ts [hunk 2]
```diff
@@ -60,7 +66,7 @@ export default class Indexer extends HealthCheck {
    * Command Entry Point
    */
   async run(): Promise<void> {
-    this.log(`Operator command has begun!!!`)
+    this.log(`Indexer command has begun!!!`)
```
Comment (sogoiii) [inline src/commands/indexer/index.ts:69]: Lol, good catch.

File: src/utils/network-monitor.ts [hunk 3]
```diff
@@ -582,6 +585,8 @@ export class NetworkMonitor {
       this.log(`📄 Interfaces address: ${this.interfacesAddress}`)
       this.log(`📄 Operator address: ${this.operatorAddress}`)
       this.log(`📄 Registry address: ${this.registryAddress}`)
+      this.log(`📄 HLG Token address: ${this.tokenAddress}`)
+      this.log(`📄 CXIP ERC721 address: ${this.cxipERC721Address}`)
```
Comment (sogoiii) [inline src/utils/network-monitor.ts]: Since we are using this address just so we can have a hardhat class, and is really an implementation detail. Should we label it? I don' t want people thinking that there is only cxipERC721Address. How do we make sure people don't make the wrong assumption?

1. `CXIP ERC721 address (internal instance)` 
2. `CXIP ERC721 address (internal)`
3. `CXIP ERC721 address (hardhat contract initializer)`

🤔

File: src/commands/propagator/index.ts [hunk 4]
```diff
@@ -26,9 +26,7 @@ type RecoveryData = {
 export default class Propagator extends HealthCheck {
   static hidden = true
   static description = 'Listen for EVM events deploys collections to the supported networks'
-  static examples = [
-    '$ <%= config.bin %> <%= command.id %> --networks ethereumTestnetGoerli polygonTestnet avalancheTestnet --mode=auto',
-  ]
+  static examples = ['$ <%= config.bin %> <%= command.id %> --networks goerli mumbai fuji --mode=auto']
```
Comment (sogoiii) [inline src/commands/propagator/index.ts:29]: nice catch.

File: src/utils/contracts.ts [hunk 5]
```diff
@@ -35,6 +35,14 @@ export const FAUCET_ADDRESSES: {[key in Environment]: string} = {
   [Environment.mainnet]: '0x0000000000000000000000000000000000000000',
 } as const
 
+export const CXIP_ERC721_ADDRESSES: {[key in Environment]: string} = {
+  [Environment.localhost]: '0x0000000000000000000000000000000000000000',
+  [Environment.experimental]: '0x0000000000000000000000000000000000000000',
+  [Environment.develop]: '0x156C8b069232986c8C6bD9017BBAA098e97f0269',
+  [Environment.testnet]: '0x0000000000000000000000000000000000000000',
```
Comment (sogoiii) [inline src/utils/contracts.ts]: Should we set an address for testnet and mainnet so we dont have to backtrack?

Comment (alexanderattar) [reply src/utils/contracts.ts]: yeah good call

File: src/utils/network-monitor.ts (hunk 3 above)
Comment (alexanderattar) [reply src/utils/network-monitor.ts]: oh yeah that's true. we don't need to log it here. it was mostly just for me while i was testing this

File: .github/workflows/cicd_develop_staging.yaml (hunk 1 above)
Comment (alexanderattar) [reply .github/workflows/cicd_develop_staging.yaml:39]: this was just a whitespace issue caught by prettier. the change shouldn't effect anything since it's just a comment

Comment (sogoiii) [reply .github/workflows/cicd_develop_staging.yaml:39]: ah ok. I see you are right about that.

**Review by sogoiii — APPROVED** [review]
👍
